internal/jimm/juju/migrationtarget.go

```diff
@@ -466,3 +489,68 @@ func (j *JujuManager) importModelFromDescription(ctx context.Context, targetCont
 	}
 	return nil
 }
+
+// CleanupPartialModelMigrations cleans up any partial model migrations that have exceeded the timeout.
+// It deletes the incoming model migration record, deletes the user mappings for the model,
+// and deletes the model record from JIMM's state.
+func (j *JujuManager) CleanupPartialModelMigrations(ctx context.Context) error {
+	const op = errors.Op("jimm.CleanupPartialModelMigrations")
+
+	// Get all incoming model migrations that have exceeded the timeout.
+	migrations, err := j.Database.GetIncomingModelMigrationsCreatedBefore(ctx, time.Now().Add(-TIMEOUT_PENDING_MIGRATION))
```

Comment:
Another option but fine with the current name,
```suggestion
	migrations, err := j.Database.GetIncomingModelMigrationsByCreatedDate(ctx, time.Now().Add(-TIMEOUT_PENDING_MIGRATION))
```